wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/RuntimeInfrastructure.java

```diff
@@ -83,55 +57,6 @@ public final EventService getEventService() {
   }
 
   /**
-   * Preliminary estimates incoming Environment. Should be used for environment validation before
-   * storing and creation of {@link RuntimeContext}.
-   *
-   * <p>It is supposed that result is used as a parameter of the method {@link
-   * #prepare(RuntimeIdentity, InternalEnvironment)}. <br>
-   * Note: this method will be eventually final, but it is not for now for workaround in Docker
-   * infra - in dockerimage environment image should be in content, not in location. It is marked
-   * with {@link Beta} annotation to hint that. Do not override this method. <br>
-   * Workaround should be removed after resolution of https://github.com/eclipse/che/issues/6006
-   *
-   * @param environment incoming Environment to estimate
-   * @return calculated internal environment.
-   * @throws ValidationException if incoming Environment is not valid
-   * @throws InfrastructureException if any other error occurred
-   */
-  @Beta
-  public InternalEnvironment estimate(Environment environment)
```

Comment:
There is no any abstract logic in this class anymore. I'd change abstract class here to interface.